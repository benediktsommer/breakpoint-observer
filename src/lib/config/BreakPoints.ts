/**
 * @type {Object}
 */
export const BreakPoints = {
  xxs: {
    width: 240,
    mediaQuery: '(max-width: 319px) and (min-width: 240px)',
  },
  xs: {
    width: 320,
    mediaQuery: '(max-width: 479px) and (min-width: 320px)',
  },
  s: {
    width: 480,
    mediaQuery: '(max-width: 767px) and (min-width: 480px)',
  },
  m: {
    width: 768,
    mediaQuery: '(max-width: 1023px) and (min-width: 768px)',
  },
  l: {
    width: 1024,
    mediaQuery: '(max-width: 1279px) and (min-width: 1024px)',
  },
  xl: {
    width: 1280,
    mediaQuery: '(max-width: 1599px) and (min-width: 1280px)',
  },
  xxl: {
    width: 1600,
    mediaQuery: '(min-width: 1600px)',
  },
};
