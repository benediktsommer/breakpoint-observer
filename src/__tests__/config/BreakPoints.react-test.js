/*global
    jest,
    console,
    describe,
    expect,
    it,
    afterEach,
    afterAll,
    beforeEach,
    beforeAll,
    require,
    process
 */
import React from 'react';
import renderer from 'react-test-renderer';

import { BreakPoints } from '../../lib/config/BreakPoints';

describe('Test BreakPoints', () => {
    describe('when the config object is called', () => {
        it('the breakpoints are given back', () => {
            expect(BreakPoints).toMatchSnapshot();
        });
    })
});
