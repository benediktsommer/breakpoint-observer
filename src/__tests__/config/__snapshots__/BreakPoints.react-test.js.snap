// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test BreakPoints when the config object is called the breakpoints are given back 1`] = `
Object {
  "l": Object {
    "mediaQuery": "(max-width: 1279px) and (min-width: 1024px)",
    "width": 1024,
  },
  "m": Object {
    "mediaQuery": "(max-width: 1023px) and (min-width: 768px)",
    "width": 768,
  },
  "s": Object {
    "mediaQuery": "(max-width: 767px) and (min-width: 480px)",
    "width": 480,
  },
  "xl": Object {
    "mediaQuery": "(max-width: 1599px) and (min-width: 1280px)",
    "width": 1280,
  },
  "xs": Object {
    "mediaQuery": "(max-width: 479px) and (min-width: 320px)",
    "width": 320,
  },
  "xxl": Object {
    "mediaQuery": "(min-width: 1600px)",
    "width": 1600,
  },
  "xxs": Object {
    "mediaQuery": "(max-width: 319px) and (min-width: 240px)",
    "width": 240,
  },
}
`;
